cpp/src/io/utilities/file_io_utilities.hpp

```diff
@@ -93,7 +93,10 @@ class cufile_input : public cufile_io_base {
    */
   virtual std::unique_ptr<datasource::buffer> read(size_t offset,
                                                    size_t size,
-                                                   rmm::cuda_stream_view stream) = 0;
+                                                   rmm::cuda_stream_view stream)
```

Comment:
These were pure virtual methods for a reason, I believe. Why change this? CC @vuule .

Reply:
When I tried to build and run CUDF in my environment, it terminated abnormally while the benchmark build was in progress, trying to virtual std::future<size_t> read_async() bind.

I tried not to build the benchmark and the build worked. However, when importting cudf s.t.

import cudf

while running Python, I got an error trying to do a virtual std::future<size_t> read_async() bind as well.

Reply:
I think we need to provide a default working implementation for `read_async`, instead of throwing. Parquet, ORC and Avro readers would not work with custom datasource implementations if we datasource::read_async always throws.

Reply:
Can you please share the compiler errors you are seeing?

Reply:
@harrism san
/usr/bin/ld: ../libcudf.so: undefined reference to `cudf::io::detail::cufile_input::read_async(unsigned long, unsigned long, unsigned char*, rmm::cuda_stream_view)'
collect2: エラー: ld はステータス 1 で終了しました (collect2: error: id ended with status 1)

Reply:
Thank you for this info. I understand the issue now. We did not implement read_async in the cufile datasource.